src/main/java/com/codeus/winter/config/DefaultBeanFactory.java

```diff
@@ -11,108 +12,143 @@
             IllegalAccessException, NoSuchMethodException {
-        uniqueBeanChecking(beanClass);
+        checkBeanNameUniqueness(beanClass);
 
         Object newBean = beanClass.getDeclaredConstructor().newInstance();
         singletonBeans.put(newBean.getClass().getName(), newBean);
         return newBean;
     }
 
+    /**
+     * Creating bean object with name.
+     * @param name bean's name
+     * @return bean object if its not possible throw exception.
+     */
     @Override
-    public Object createBean(String name) throws NotUniqueBeanDefinitionException {
-        uniqueBeanChecking(name);
+    public Object createBean(@Nonnull final String name) throws NotUniqueBeanDefinitionException {
+        checkBeanNameUniqueness(name);
 
         Object newBean = new Object();
         singletonBeans.put(name, newBean);
         return newBean;
     }
 
+    /**
+     * Register bean in the bean's storage.
+     * @param name bean's name.
+     * @param beanDefinition bean's BeanDefinition.
+     * @param beanInstance bean's instance.
+     */
     @Override
-    public void registerBean(String name, BeanDefinition beanDefinition, Object beanInstance) {
+    public void registerBean(@Nonnull final String name,
+                             @Nonnull final BeanDefinition beanDefinition,
+                             @Nonnull final Object beanInstance) {
         if (beanDefinition.isSingleton()) {
             singletonBeans.put(name, beanInstance);
         }
         beanDefinitions.put(name, beanDefinition);
     }
 
+    /**
+     * Adding BeanPostProcessor to the storage.
+     *
+     * @param postProcessor BeanPostProcessor.
+     */
     @Override
-    public void addBeanPostProcessor(BeanPostProcessor postProcessor) {
+    public void addBeanPostProcessor(@Nonnull final BeanPostProcessor postProcessor) {
         postProcessors.add(postProcessor);
     }
 
-    private void uniqueBeanChecking(String beanName) {
-        var isNotUniq = singletonBeans.entrySet().stream()
-                .map(entry -> entry.getKey().equals(beanName))
-                .anyMatch(beanName::equals);
-        if (isNotUniq) {
-            throw new NotUniqueBeanDefinitionException("Bean '" + beanName + "' already exists");
+    private void checkBeanNameUniqueness(@Nonnull final String beanName) {
+        if (singletonBeans.containsKey(beanName)) {
+            throw new NotUniqueBeanDefinitionException(
+                    String.format("Bean with name '%s' already exists", beanName));
         }
     }
 
-    private void uniqueBeanChecking(Class<Object> beanClass) {
-        for (String key : singletonBeans.keySet()) {
-            if (beanClass.isInstance(singletonBeans.get(key))) {
-                throw new NotUniqueBeanDefinitionException("Bean '" + beanClass.getName() + "' already exists");
-            }
+    private void checkBeanNameUniqueness(@Nonnull final Class<Object> beanClass) {
```

Comment:
Thanks for accepting my suggestion :) 
In the meantime, could you please rename only this method? It checks bean uniqueness, not the name itself.

Reply:
`checkBeanUniqueness` would be good enough or any other options of your choice.